apps/api/src/expenses/expenses.controller.ts

```diff
@@ -46,11 +62,60 @@ export class ExpensesController {
   }
 
   @Delete(':id')
+  remove(@Param('id') id: string) {
+    return this.expensesService.remove(id)
+  }
+
+  @Get('campaign/:slug')
   @Roles({
     roles: [RealmViewSupporters.role, ViewSupporters.role],
     mode: RoleMatchingMode.ANY,
   })
-  remove(@Param('id') id: string) {
-    return this.expensesService.remove(id)
+  async listCampaignExpenses(@Param('slug') slug: string) {
+    return this.expensesService.listCampaignExpenses(slug)
+  }
+
+  @Get('campaign/approved/:slug')
+  @Public()
+  async listCampaignApprovedExpenses(@Param('slug') slug: string) {
+    return this.expensesService.listCampaignApprovedExpenses(slug)
+  }
+
+  @Post('upload-files/:id')
+  @UseInterceptors(FilesInterceptor('file', 5, { limits: { fileSize: 10485760 } })) //limit uploaded files to 5 at once and 10MB each
+  @Roles({
+    roles: [RealmViewSupporters.role, ViewSupporters.role],
+    mode: RoleMatchingMode.ANY,
+  })
+  uploadFiles(@Param('id') id: string, @UploadedFiles() files: Express.Multer.File[]) {
+    return this.expensesService.uploadFiles(id, files)
+  }
+
+  @Get('files/:id')
+  getUploadedFiles(@Param('id') id: string) {
+    return this.expensesService.listUploadedFiles(id)
+  }
+
+  @Get('download-file/:fileId')
+  @Public()
+  async downloadFile(
+    @Param('fileId') fileId: string,
+    @Response({ passthrough: true }) res,
+  ): Promise<StreamableFile> {
```

Comment:
endpoint should be file/:fileId

Reply:
I think download-file is better here. It is more verbose and explains that we are fetching the actual file's content, not the expense_file db record.